node/src/index.ts

```diff
@@ -258,14 +278,8 @@ export class LocalConnection implements Connection {
   async openTable<T> (name: string, embeddings?: EmbeddingFunction<T>): Promise<Table<T>> {
     // TODO: move this thing into rust
     const callArgs = [this._db, name]
-    const awsCredentials = this._options().awsCredentials
-    if (awsCredentials !== undefined) {
-      callArgs.push(awsCredentials.accessKeyId)
-      callArgs.push(awsCredentials.secretKey)
-      if (awsCredentials.sessionToken !== undefined) {
-        callArgs.push(awsCredentials.sessionToken)
-      }
-    }
+    appendAwsCreds(callArgs, this._options().awsCredentials, this._options().awsRegion)
```

Comment:
why have side-effects as part of the method as opposed to having the call site here concatenate the callArgs and the output?

Reply:
ah, already fixed this. Passing array back and expanding them now